ckanext/ldap/controllers/user.py

```diff
@@ -292,18 +292,26 @@ def _find_ldap_user(login):
         attributes.append(config[u'ckanext.ldap.fullname'])
     if u'ckanext.ldap.email' in config:
         attributes.append(config[u'ckanext.ldap.email'])
+
     try:
-        ret = _ldap_search(cnx, filter_str, attributes, non_unique=u'log')
+        ret = _ldap_search(cnx, filter_str, attributes, config[u'ckanext.ldap.base_dn'], non_unique=u'log')
         if ret is None and u'ckanext.ldap.search.alt' in config:
             filter_str = config[u'ckanext.ldap.search.alt'].format(
                 login=ldap.filter.escape_filter_chars(login))
-            ret = _ldap_search(cnx, filter_str, attributes, non_unique=u'raise')
+            ret = _ldap_search(cnx, filter_str, attributes, config[u'ckanext.ldap.base_dn'], non_unique=u'raise')
+            if ret is None and u'ckanext.ldap.base_dn_alt' in config:
+                filter_str = config[u'ckanext.ldap.search.filter'].format(
+                login=ldap.filter.escape_filter_chars(login))
+                ret = _ldap_search(cnx, filter_str, attributes, config[u'ckanext.ldap.base_dn_alt'], non_unique=u'raise')
+        elif ret is None and u'ckanext.ldap.base_dn_alt' in config:
+            ret = _ldap_search(cnx, filter_str, attributes, config[u'ckanext.ldap.base_dn_alt'], non_unique=u'log')
```

Comment:
If I'm reading this right, the logic here doesn't process all possible combinations of base DN and search. We try:

- `base_dn` + `search` on line 297
- `base_dn` + `search.alt`: on line 301
- `base_dn_alt` + `search` on line 305 and 307.

but not `base_dn_alt` + `search.alt`.

Reply:
Yes, you are correct. I have changed this code. I created a variable calles filter_str_alt in order to make the code more clear. See if you like it.
Will change my repository and should I make a new pull request?

Reply:
Looks much neater, nice work!